web/src/stream_popover.js

```diff
@@ -514,4 +514,15 @@ export function register_click_handlers() {
 
         e.stopPropagation();
     });
+
+    $("#streamlist-toggle-button").on("click", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        const sidebarHidden = !$(".app-main .column-left").hasClass("expanded");
+        popovers.hide_all();
+        if (sidebarHidden) {
+            show_streamlist_sidebar();
+        }
+    });
```

Comment:
This is progress but all code related to `show_streamlist_sidebar` might be better moved to a `left_sidebar_ui.js` module that can have logic about showing/hiding the whole left sidebar, and also any miscellaneous things that don't belong in smaller modules -- maybe a good #frontend discussion to have.